src/main/java/com/epam/preprod/sirenko/SafeContainer.java

```diff
@@ -0,0 +1,398 @@
+package com.epam.preprod.sirenko;
+
+import com.epam.preprod.sirenko.entity.Product;
+
+import java.util.*;
+
+/**
+ * Resizable array-implementation of the List interface
+ *
+ * @author E.Sirenko
+ **/
+public class SafeContainer<E extends Product> implements List<E> {
+	private Object[] array;
+	private static final String EXCEPTION_INFO_INDEX = "Index: ";
+	private static final String EXCEPTION_INFO_SIZE = ", Size: ";
+	
+	/**
+	 * Constructs an empty list with capacity of ten.
+	 **/
+	public SafeContainer() {
+		this.array = new Object[]{};
+	}
+	
+	/**
+	 * Returns the number of elements
+	 **/
+	@Override
+	public int size() {
+		return array.length;
+	}
+	
+	/**
+	 * Removes all the elements
+	 **/
+	@Override
+	public void clear() {
+		Object[] objects = Arrays.copyOf(array, array.length + 1);
```

Comment:
We dont need it here.We can just create new array(what is done currently in 41 line)